packages/wonder-blocks-form/src/components/field-heading.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import * as React from "react";
+import {StyleSheet} from "aphrodite";
+
+import {View} from "@khanacademy/wonder-blocks-core";
+import Color from "@khanacademy/wonder-blocks-color";
+import {Strut} from "@khanacademy/wonder-blocks-layout";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {
+    type Typography,
+    LabelMedium,
+    LabelSmall,
+} from "@khanacademy/wonder-blocks-typography";
+
+type Props = {|
+    /**
+     * The form field component.
+     */
+    field: React.Node,
+
+    /**
+     * The title for the label element.
+     */
+    label: string | React.Element<Typography>,
+
+    /**
+     * The text for the description element.
+     */
+    description?: string | React.Element<Typography>,
+
+    /**
+     * The message for the error element.
+     */
+    error?: string | React.Element<Typography>,
+|};
+
+export default class FieldHeading extends React.Component<Props> {
+    isString: (item: string | React.Element<Typography>) => boolean = (
+        item,
+    ) => {
+        // Helps check if a prop is of type "string" to either render the default
+        // label or a custom Typography label passed by the user.
+        return typeof item === "string";
+    };
+
+    render(): React.Node {
```

Comment:
As we are going to generate IDs to connect the label with the field, it would be good to use `IDProvider` here to handle that logic for us.

You can see an example here: https://deploy-preview-1140--wonder-blocks.netlify.app/#!/IDProvider/1